uiowa/uiowa
# brand.uiowa.edu site split export as v3

relates to https://github.com/uiowa/uiowa/issues/5203

The idea behind this is to allow brand.uiowa.edu to continue as a v2 site in PROD while a v3 site is in stage while maintaining their custom functionality. We cannot have import errors or loss of functionality whether main is deployed to stage or prod.

## To Test

- checkout brand_v3
- for good measure `ddev composer install && ddev blt frontend`
- If you have the brand site local, `ddev drush @brand.local sql-drop && ddev drush @brand.local cr`
- First try and sync the PROD site to see if there are any import issues: `ddev blt ds --site=brand.uiowa.edu`
- As a webmaster, poke around the site making sure you can't see layout builder or page replicate functionality but can see the lockup functionality.
- Run `ddev drush @brand.local cr` and then try to export the site `ddev drush @brand.local config-split:export site` and review, if any, differences.
  - Disregard and remove the following (`git clean -fd config/sites/brand.uiowa.edu/`):
    - config/sites/brand.uiowa.edu/system.action.user_add_role_action.webmaster.yml
    - config/sites/brand.uiowa.edu/system.action.user_remove_role_action.webmaster.yml
    - config/sites/brand.uiowa.edu/user.role.webmaster.yml
- Next try to install brand as a fresh v3 site `ddev drush @brand.local sql-drop && ddev drush @brand.local cr && ddev blt drupal:install --site=brand.uiowa.edu` enable some modules `ddev drush @brand.local en mailsystem swiftmailer workbench_email brand_core` and then run the site config import `ddev drush @brand.local config:import --source ../config/sites/brand.uiowa.edu --partial` followed by `ddev drush @brand.local cim` and look for any import errors.
- Poke around the site as a webmaster (make an account with webmaster role to masquerade with) and make sure you can do layout builder type stuff along with the lockup related stuff.
- Run `ddev drush @brand.local cr` and then try to export the site `ddev drush @brand.local config-split:export site` and review, if any, differences.
  - Disregard and remove the following (`git clean -fd config/sites/brand.uiowa.edu/`):
    - config/sites/brand.uiowa.edu/system.action.user_add_role_action.webmaster.yml
    - config/sites/brand.uiowa.edu/system.action.user_remove_role_action.webmaster.yml
    - config/sites/brand.uiowa.edu/user.role.webmaster.yml
